lib/src/room.dart

```diff
@@ -1187,22 +1187,18 @@ class Room {
   /// Set the power level of the user with the [userID] to the value [power].
   /// Returns the event ID of the new state event. If there is no known
   /// power level event, there might something broken and this returns null.
-  Future<String> setPower(String userID, int power) async {
-    final powerMap = Map<String, Object?>.from(
-      getState(EventTypes.RoomPowerLevels)?.content ?? {},
-    );
-
-    final usersPowerMap = powerMap['users'] is Map<String, Object?>
-        ? powerMap['users'] as Map<String, Object?>
-        : (powerMap['users'] = <String, Object?>{});
+  Future<String> setPower(String userId, int power) async {
+    final powerLevelMapCopy =
+        getState(EventTypes.RoomPowerLevels)?.content.copy() ?? {};
 
-    usersPowerMap[userID] = power;
+    powerLevelMapCopy['users'] ??= <String, Object?>{};
+    powerLevelMapCopy.tryGetMap<String, Object?>('users')?[userId] = power;
 
     return await client.setRoomStateWithKey(
```

Comment:
this actually now means that the changes will only be reflected in the state event later. I would bet there is some code that depends on this function having immediately reflected the changes. Should we consider this a breaking change or maybe wait for the event to come down in sync? Maybe we can just ignore that, but I think it makes sense to consider the impact of that. For example the following code would now only set the permissions for one of the users:

```dart
await room.setPower('@a:example.org', 10);
await room.setPower('@b:example.org', 10);
```

Reply:
We discussed this and agreed that just documenting it is the best solution for now.